### DISCUSSION TITLE
low-memory

### DISCUSSION THREAD

**Discussion by Dan-Burns** [chaidiscovery/chai-lab#266] [General]
Hi,

I've have a GPU with 24G and have been able to predict structures over 700 aa. I tried --low-memory on a 1200 aa structure but it runs out of memory. But I believe low-memory is the reason I was able to get a 900 aa prediction. 

Has anybody explored how the scaling works for maximum sequence length and memory usage between normal inference and with the --low-memory flag? 

Does using ESM embeddings significantly affect memory usage? I'm also interested in your thoughts on whether these embeddings improve the output - from limited experience and simply by the looks of it, it seems to.

Thanks,
Dan

**Comment by arogozhnikov** [comment]
> Does using ESM embeddings significantly affect memory usage? 

it doesn't affect memory at all (unless you can't run it, but with 24GB you're ok)

> I'm also interested in your thoughts on whether these embeddings improve the output - from limited experience and simply by the looks of it, it seems to.

yes, they improve results; specially when mining relevant MSAs is hard

`low_memory=True` cleans up maybe 25% of memory (don't expect magic there) and is used by default
